components/script/script_thread.rs

```diff
@@ -1044,7 +1209,20 @@ impl ScriptThread {
         // Process the gathered events.
         debug!("Processing events.");
         for msg in sequential {
+            if self.tti_window.borrow().needs_check() {
+                if !self.check_tti() {
+                    self.tti_window.borrow_mut().start_window();
+                } else {
+                    // all current documents are interactive
+                    self.tti_window.borrow_mut().set_all_interactive();
+                }
+            }
             debug!("Processing event {:?}.", msg);
+
+            // TODO avada currently removing completed tasks
+            // even though i'm tracking end times, i'm not keeping the task after it's been completed
+            let mut completed = self.task_times.borrow_mut().remove(0); //need to pop front
+            completed.complete();
```

Comment:
We're marking the task as complete, but we haven't actually run it yet? I don't understand the reasoning here.